src/OpenLoco/Paint/Paint.h

```diff
@@ -0,0 +1,158 @@
+        coord_t map_x; // 0x2C
+        coord_t map_y; // 0x2E
+        union
+        {
+            Map::tile_element* tileElement; // 0x30 (or thing pointer)
+            Thing* thing;                   // 0x30
+        };
+    };
+    static_assert(sizeof(PaintStruct) == 0x34);
+
+    union PaintEntry
+    {
+        PaintStruct basic;
+        AttachedPaintStruct attached;
+        PaintStringStruct string;
+    };
+    static_assert(sizeof(PaintEntry) == 0x34);
+#pragma pack(pop)
+
+    struct PaintSession
+    {
+    public:
+        void generate();
+        void arrangeStructs();
+        [[nodiscard]] Ui::ViewportInteraction::InteractionArg getNormalInteractionInfo(const uint32_t flags);
+        [[nodiscard]] Ui::ViewportInteraction::InteractionArg getStationNameInteractionInfo(const uint32_t flags);
+        [[nodiscard]] Ui::ViewportInteraction::InteractionArg getTownNameInteractionInfo(const uint32_t flags);
+
+    private:
+        //Gfx::drawpixelinfo_t* dpi;                     // 0xE0C3E0
+        //PaintEntry paintStructs[4000];                 // 0xE0C410 -> 0xE3F050
+        //PaintStruct* quadrants[1024];                  // 0xE3F0C0 -> 0xE400C0
+        //uint32_t quadrantBackIndex;                    // 0xE400C0
+        //uint32_t quadrantFrontIndex;                   // 0xE400C4
+        //const void* currentlyDrawnItem;                // 0xE4F0B4
+        //PaintEntry* endOfPaintStructArray;             // 0xE0C404
+        //PaintEntry* nextFreePaintStruct;               // 0xE0C40C
+        //Map::map_pos spritePosition;                   // 0xE3F090 and 0xE3F096
+        //PaintStruct* lastRootPS;                       // 0xE40120
+        //UI::ViewportInteraction::InteractionItem type; // 0xE3F0AC
+        //PaintStringStruct* paintStringHead;            // 0xE40118
+        //PaintStringStruct* lastPaintString;            // 0xE4011C
+        //Map::map_pos mapPosition;                      // 0xE3F0B0
+
+        // From OpenRCT2 equivalent fields not found yet or new
+        //uint8_t currentRotation;                     // new field
+        //PaintStruct paintHead;                       // new field
+        //uint32_t viewFlags;                          // new field might not be needed tbc
+        //AttachedPaintStruct* unkF1AD2C;              // no equivalent
+        //support_height supportSegments[9];
+        //support_height support;
+        //PaintStruct* woodenSupportsPrependTo;
+        //tunnel_entry leftTunnels[TUNNEL_MAX_COUNT];
+        //uint8_t leftTunnelCount;
+        //tunnel_entry rightTunnels[TUNNEL_MAX_COUNT];
+        //uint8_t rightTunnelCount;
+        //uint8_t verticalTunnelHeight;
+        //const Map::tile_element* surfaceElement;
+        //Map::tile_element* pathElementOnSameHeight;
+        //Map::tile_element* trackElementOnSameHeight;
+        //bool didPassSurface;
+        //uint8_t unk141E9DB;
+        //uint16_t waterHeight;
+        //uint32_t trackColours[4];
```

Comment:
Should these be included, if they are not obviously present?

Reply:
pretty sure the tunnels are included and I'm pretty sure the supports are also included. I'd rather keep them in for now then we can remove them as we go.